File: Lesson09_OOP/02_NotEncapsulated/Clothing.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class Clothing {

    public String colour;
    public String material;

    // Getters and setters
    public String getColour() { return this.colour; }
    public String getMaterial() { return this.material; }
    public void setColour(String colour) { this.colour = colour; }
    public void setMaterial(String material) { this.material = material; }

    // Constructor
    public Clothing(String colour, String material) {
        this.colour = colour;
        this.material = material;
    }

    public static void main(String[] args) {
        var garment = new Clothing("black", "cotton");
        garment.colour = "yellow";
        garment.material = "bamboo";
        System.out.printf("Colour: %s, Material: %s\n", 
          garment.colour, garment.material);
    }
}

File: Lesson09_OOP/01_Encapsulation/Clothing.py

"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""

class Clothing:
    def __init__(self, colour, material):
        self.colour = colour
        self.material = material

    @property
    def colour(self):
        return self._colour

    @colour.setter
    def colour(self, value):
        self._colour = value

    @property
    def material(self):
        return self._material

    @material.setter
    def material(self, value):
        self._material = value

# Example usage
if __name__ == "__main__":
    garment = Clothing("black", "cotton")
    print(f"Colour: {garment.colour}, Material: {garment.material}")
